### PR TITLE
Add functionality to fetch additional package info from a web page.

### PR DESCRIPTION
**Pull request by huguesv**
Add missing fields to RPackage.

### CODE REVIEW

File: src/R/Components/Impl/PackageManager/Implementation/RPackageWebParser.cs [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+        /// won't be overwritten by the ones extracted from the html page.
+        /// </remarks>
+        public static async Task<RPackage> RetrievePackageInfo(Uri packageUri, RPackage result = null) {
+            string content;
+            try {
+                using (var webClient = new WebClient()) {
+                    content = await webClient.DownloadStringTaskAsync(packageUri);
+                }
+            } catch (WebException ex) {
+                throw new RPackageInfoRetrievalException(string.Format("Error reading index page from {0}", packageUri), ex);
```
Comment (AlexanderSher) [inline src/R/Components/Impl/PackageManager/Implementation/RPackageWebParser.cs]: Who is going to catch this exception? RPackageManager or view model?

File: src/R/Components/Impl/PackageManager/Implementation/RPackageWebParser.cs [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+            public const string License = "License";
+            public const string Version = "Version";
+            public const string NeedsCompilation = "NeedsCompilation";
+            public const string Author = "Author";
+            public const string URL = "URL";
+            public const string LinkingTo = "LinkingTo";
+            public const string Enhances = "Enhances";
+            public const string Maintainer = "Maintainer";
+            public const string BugReports = "BugReports";
+            public const string Published = "Published";
```
Comment (AlexanderSher) [inline src/R/Components/Impl/PackageManager/Implementation/RPackageWebParser.cs]: Looks like you can use `nameof` for every field

File: src/R/Components/Impl/PackageManager/Implementation/RPackageWebParser.cs (hunk 1 above)
Comment (huguesv) [inline src/R/Components/Impl/PackageManager/Implementation/RPackageWebParser.cs]: Intent was that the caller of RPackageManager, so yes probably view model.  The alternative was that RPackageManager catches it, but then view model wouldn't know things went wrong and would have no way of indicating that to the user.

File: src/R/Components/Impl/PackageManager/Implementation/RPackageWebParser.cs [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+                if (string.IsNullOrEmpty(_pkg.Description)) {
+                    _pkg.Description = GetElementText(paragraphNode);
+                }
+            }
+
+            private void ParseGenericField(ElementNode tdNode1, ElementNode tdNode2) {
+                string fieldName = GetElementText(tdNode1).Trim(':');
+                string fieldValue = GetElementText(tdNode2);
+
+                switch (fieldName) {
```
Comment (AlexanderSher) [inline src/R/Components/Impl/PackageManager/Implementation/RPackageWebParser.cs]: I don't like this switch, but can't think of anything better :)

File: src/R/Components/Impl/PackageManager/Implementation/RPackageWebParser.cs (hunk 1 above)
Comment (MikhailArkhipov) [inline src/R/Components/Impl/PackageManager/Implementation/RPackageWebParser.cs]: Do we need to show some sort of progress and or cancellation? If connection is offline, timeout, I think, is 30 seconds. Or maybe check connection before trying actual download?

File: src/R/Components/Impl/PackageManager/Implementation/RPackageWebParser.cs [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+            }
+
+            var pkg = result ?? new RPackage();
+            try {
+                var tree = new HtmlTree(new Microsoft.Web.Core.Text.TextStream(content));
+                tree.Build();
+
+                var search = new IndexVisitor(pkg);
+                tree.Accept(search, null);
+            } catch (Exception ex) when(!ex.IsCriticalException()) {
```
Comment (MikhailArkhipov) [inline src/R/Components/Impl/PackageManager/Implementation/RPackageWebParser.cs]: There is nothing that is supposed to throw there. The parser is the editor-style recoverable parser so it handles broken HTML.

Comment (MikhailArkhipov) [inline src/R/Components/Impl/PackageManager/Implementation/RPackageWebParser.cs]: Although resulting tree may not be what you'd expect. The parser does not do syntax check, in  HTML editor there is a separate validator module.

File: src/R/Components/Impl/PackageManager/Implementation/RPackageWebParser.cs (hunk 3 above)
Comment (huguesv) [inline src/R/Components/Impl/PackageManager/Implementation/RPackageWebParser.cs]: I'll try with reflection, and see if performance is not too affected.

Comment (MikhailArkhipov) [inline src/R/Components/Impl/PackageManager/Implementation/RPackageWebParser.cs]: Nah, leave that alone IMO

File: src/R/Components/Impl/PackageManager/Implementation/RPackageWebParser.cs (hunk 4 above)
Comment (huguesv) [inline src/R/Components/Impl/PackageManager/Implementation/RPackageWebParser.cs]: I'll remove the try/catch then.

**Comment by MikhailArkhipov** [comment]
:+1:

**Pull request merged**